pkg/reconciler/mutatingwebhook/mutatingwebhook_controller.go

```diff
@@ -0,0 +1,81 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"k8s.io/api/admissionregistration/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/logger"
+)
+
+var log = logger.New("mutatingwebhook-reconciler")
+
+// MutatingWebhookConfigrationReconciler reconciles a MutatingWebhookConfiguration object
+type MutatingWebhookConfigrationReconciler struct {
+	client.Client
+	Scheme       *runtime.Scheme
+	OsmWebhook   string
+	OsmNamespace string
+	CertManager  certificate.Manager
+}
+
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=mutatingwebhookconfigurations,verbs=get;list;watch;update;
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=mutatingwebhookconfigurations/status,verbs=get;update;patch
```

Comment:
Why is this required?

Reply:
we are using kubebuilder to test the controllers, and these lines add the necessary permissions in kubebuilders to listen on events related to muutatingwebhooks

Reply:
I don't understand this since I am not familiar with kubebuilder, but if you think a few comments on this could help readability they would be welcome.

Reply:
I'm not sure what comments I can add, but do you think a reference to a doc would help ?